pkg/controller/controller_instance_test.go

```diff
@@ -1136,6 +1225,133 @@ func TestPollServiceInstanceSuccessDeprovisioningWithOperationWithFinalizer(t *t
 	assertNumEvents(t, events, 1)
 }
 
+func TestSetInstanceCondition(t *testing.T) {
```

Comment:
fwiw, that works for me. I would be fine with you putting that comment in following this PR, so that it doesn't block it